pkg/cmd/cli/cmd/newapp.go

```diff
@@ -419,22 +420,26 @@ func setAppConfigLabels(c *cobra.Command, config *newcmd.AppConfig) error {
 	return nil
 }
 
-func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args []string, config *newcmd.AppConfig) error {
-	namespace, _, err := f.DefaultNamespace()
-	if err != nil {
-		return err
-	}
-
+// getDockerClient returns a client capable of communicating with the local
+// docker daemon.  If an error occurs (such as no local daemon being available),
+// it will return nil.
+func getDockerClient() (*docker.Client, error) {
 	dockerClient, _, err := dockerutil.NewHelper().GetClient()
 	if err == nil {
-		if err = dockerClient.Ping(); err == nil {
-			config.SetDockerClient(dockerClient)
```

Comment:
note i've completely eliminated this now. we always just go through the SetOpenShiftClient path (which is what was effectively happening before anyway, so this is really just removing dead code.  The only question is whether it was intentionally dead code)